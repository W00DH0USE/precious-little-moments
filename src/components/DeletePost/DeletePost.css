.delete-post-form {
  background-color: #fff;
  height: 350px;
}

.delete-post-title {
  font-size: 24px;
}

.delete-warning {
  color: red;
  font-size: 15px;
}