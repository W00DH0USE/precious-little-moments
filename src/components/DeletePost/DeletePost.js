import React from 'react';
import { Form, Button } from "../Utils/Utils";
import './DeletePost.css';

function DeletePost(props) {
  const post = props.posts.filter(post => post.id === props.postId)
  return (
    <div className='modal-wrapper deletePost'>
      <div className="modal-content">
        <Form className='delete-post-form' onSubmit={props.handleDelete}>
          <div className="delete-post-form-title">
            <h2 className="Form__title">Precious Little Moment: <span><em>delete moment</em></span></h2>
            <hr />
            <h3 className='delete-post-title'>Delete Moment "{post[0].post_title}"?</h3>
            <p className="delete-warning">*Please make sure you would like to delete this moment as this cannot be recovered!*</p>
          </div>
          <Button className="button demo-button button-cancel" type="submit">Delete</Button>
          <Button className="button demo-button button-continue" onClick={props.close}>Cancel</Button>
        </Form>
      </div>
    </div>
  )
}

export default DeletePost;