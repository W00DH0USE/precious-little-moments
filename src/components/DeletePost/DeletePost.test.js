import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DeletePost from './DeletePost';


describe(`Delete Post Component`, () => {
  const post = [
    {
      post_title: 'Test Post Title 1',
      post_content: 'Test Post content 1',
      owner: 1
    },
    {
      post_title: 'Test Post Title 2',
      post_content: 'Test Post content 2',
      owner: 1
    },
    {
      post_title: 'Test Post Title 3',
      post_content: 'Test Post content 3',
      owner: 1
    },
  ]
  it('renders without crashing', () => {
    const wrapper = shallow(<DeletePost posts={post} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});