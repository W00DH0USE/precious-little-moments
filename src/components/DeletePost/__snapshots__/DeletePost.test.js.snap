// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Post Component renders without crashing 1`] = `
<div
  className="modal-wrapper deletePost"
>
  <div
    className="modal-content"
  >
    <Form
      className="delete-post-form"
    >
      <div
        className="delete-post-form-title"
      >
        <h2
          className="Form__title"
        >
          Precious Little Moment: 
          <span>
            <em>
              delete moment
            </em>
          </span>
        </h2>
        <hr />
        <h3
          className="delete-post-title"
        >
          Delete Moment "
          Test Post Title 1
          "?
        </h3>
        <p
          className="delete-warning"
        >
          *Please make sure you would like to delete this moment as this cannot be recovered!*
        </p>
      </div>
      <Button
        className="button demo-button button-cancel"
        type="submit"
      >
        Delete
      </Button>
      <Button
        className="button demo-button button-continue"
      >
        Cancel
      </Button>
    </Form>
  </div>
</div>
`;
